File: library/src/main/java/io/vithor/yamvpframework/migration/V1_TestMigration.java
package io.vithor.yamvpframework.migration;

import android.database.SQLException;
import android.database.sqlite.SQLiteOpenHelper;
import android.support.annotation.NonNull;

import com.j256.ormlite.support.ConnectionSource;

/**
 * Created by Vithorio Polten on 2/23/16.
 */
public class V1_TestMigration extends Migration {
    @NonNull
    @Override
    public String getDescription() {
        return "";
    }

    @Override
    public int getVersion() {
        return 1;
    }

    @Override
    public boolean doUpgrade(@NonNull SQLiteOpenHelper db, @NonNull ConnectionSource connectionSource) {
//        try {
//            TableUtils.createTableIfNotExists(connectionSource, MyTestModel.class);

//            ((DatabaseHelper)db).getMyTestModelDao().executeRaw("ALTER TABLE `mytestmodel` ADD COLUMN aloha INTEGER;");
//        } catch (SQLException e) {
//            Log.e(getClass().getName(), "Failed upgrading for migration: V" + getVersion(), e);
//            return false;
//        }
        return false;
    }
}

File: migration/src/main/java/io/vithor/yamvpframework/migration/V1_TestMigration.kt
package io.vithor.yamvpframework.migration

import android.database.sqlite.SQLiteOpenHelper
import com.j256.ormlite.support.ConnectionSource

/**
 * Created by Vithorio Polten on 2/23/16.
 */
class V1_TestMigration : Migration() {
    override val description: String
        get() = ""

    override val version: Int
        get() = 1

    override fun doUpgrade(db: SQLiteOpenHelper,connectionSource: ConnectionSource): Boolean {
        //        try {
        //            TableUtils.createTableIfNotExists(connectionSource, MyTestModel.class);

        //            ((DatabaseHelper)db).getMyTestModelDao().executeRaw("ALTER TABLE `mytestmodel` ADD COLUMN aloha INTEGER;");
        //        } catch (SQLException e) {
        //            Log.e(getClass().getName(), "Failed upgrading for migration: V" + getVersion(), e);
        //            return false;
        //        }
        return false
    }
}
